test/regression/cases/docker_containers_cpu/experiment.yaml

```diff
@@ -0,0 +1,35 @@
+optimization_goal: cpu
+
+target:
+  name: datadog-agent
+  cpu_allotment: 4
+  memory_allotment: 512 MiB
+
+  environment:
+    DD_API_KEY: a0000001
+    DD_HOSTNAME: smp-regression
+
+  profiling_environment:
+    DD_INTERNAL_PROFILING_BLOCK_PROFILE_RATE: 10000
+    DD_INTERNAL_PROFILING_CPU_DURATION: 1m
+    DD_INTERNAL_PROFILING_DELTA_PROFILES: true
+    DD_INTERNAL_PROFILING_ENABLED: true
+    DD_INTERNAL_PROFILING_ENABLE_GOROUTINE_STACKTRACES: true
+    DD_INTERNAL_PROFILING_MUTEX_PROFILE_FRACTION: 10
+    DD_INTERNAL_PROFILING_PERIOD: 1m
+    DD_INTERNAL_PROFILING_UNIX_SOCKET: /smp-host/apm.socket
+    DD_PROFILING_EXECUTION_TRACE_ENABLED: true
+    DD_PROFILING_EXECUTION_TRACE_PERIOD: 1m
+    DD_PROFILING_WAIT_PROFILE: true
+    DD_APM_INTERNAL_PROFILING_ENABLED: true
+
+    DD_INTERNAL_PROFILING_EXTRA_TAGS: experiment:docker_containers
+
+checks:
+  # This check isn’t about agent’s performance.
+  # It’s about validating that the experiment setup is correct.
+  - name: fake_check run
+    description: Ensure the agent properly detected the containers
+    bounds:
+      series: target.fake_check.run
+      lower_bound: "400"
```

Comment:
Writing this out since it took me a second:
We have 200 containers, each container configures 2 check instances to run. Each check instance emits this counter with a value of 1 on each check interval, so at check interval 1, counter will have value 400, at check interval 2, counter will have value 800, etc.
So asserting a lower-bound of 400 means that on the first check interval, all instances ran successfully.

May be worth a comment for others since its a bit subtle.

Reply:
Comment added in https://github.com/DataDog/datadog-agent/pull/33620/commits/2e40f0a6961d466316c43ddb82f744aabe763cb8.